ops/model.py

```diff
@@ -161,7 +167,13 @@ def storages(self) -> 'StorageMapping':
 
     @property
     def pod(self) -> 'Pod':
-        """Use ``model.pod.set_spec`` to set the container specification for Kubernetes charms."""
+        """Represents the definition of a pod spec in legacy Kubernetes models.
+
+        DEPRECATED: New charms should use the sidecar pattern with Pebble.
```

Comment:
should we `logging.warn` about deprecations?

Reply:
I think that's probably a bit heavy-handed, because it means that older, non-sidecar charms which have to use this get these deprecation warnings (that the user can't do anything about). That's assuming they rebuild with a recent version of ops.